src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -362,7 +346,7 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         this.createNotebookControllers(connections);
 
         // If we're listing Python kernels & there aren't any, then add a placeholder for `Python` which will prompt users to install python
-        if ('listLocalNonPythonKernels' in options && !options.listLocalNonPythonKernels) {
+        if ('useCache' in options) {
```

Comment:
why would we only care about this when using the cached kernels? It seems like it would be more necessary without the cache since the values are more likely to change.

Reply:
ah, it looks like 'useCache' is not actually what I would expect

Reply:
I get it now, I missed the | in the parameter list. 'useCache' existing in the options implies we're looking for local kernels.
Seems like this function could be split into findLocal and findRemote with how little code is shared between the two